File: 179. Largest Number.js
/**
 * @param {number[]} nums
 * @return {string}
 */
var largestNumber = function(nums) {
    let res = nums.sort((a,b) => a+''+b > b+''+a ? 1 : -1).reverse().join('');
    return res[0] == '0' ? '0' : res;
};

File: 179. Largest Number.py
def compare(a, b):
    if a+b > b+a:
        return 1
    elif a+b < b+a:
        return -1
    else:
        return 0
            
class Solution:
    # @param {integer[]} nums
    # @return {string}
    def largestNumber(self, nums):
        nums = sorted(map(str, nums), cmp = compare, reverse = True)
        return '0' if nums[0] == '0' else ''.join(nums)
